### ISSUE TITLE
Filter-Rule in Final-Unit-Indication ?

### ISSUE THREAD

**Issue by dcdolson** [lbertz02/rfc4006bis#21]
Should Final-Unit-Indication contain  * [ Filter-Rule ]
This seems to be indicated in the new text in section 8.35.

Also, if both Filter-Rule and Restriction-Filter-Rule appear in Final-Unit-Indication, do we need to explain how the precedence is to be interpreted?

**Comment by yuvalif** [comment]
we cannot add Filter-Rule to Final-Unit-Indication (not extensible). This is why we added the QoS-Final-Unit-Indication.
note that section 8.35 is not the description of the FUI AVP. it is the description of the action and apply to both new and existing AVPs. maybe, better to add that to the text:

> The access device MUST restrict the user access according to the IP
   packet filters defined in the Restriction-Filter-Rule AVP, 
   according to the IP packet filters identified by the Filter-Id AVP, or the Filter-Rule
   AVP in case of QoS-Final-Unit-Indication is used.
   All the packets not matching the filters MUST be
   dropped (see Section 5.6.3).  In the case that Restriction-Filter-
   Rule AVP is used, all the packets matching the filters MUST be
   allowed.  In the case that Filter-Rule AVP or Filter-Id AVP are used,
   other actions (e.g., QoS) may be deployed on the packets matching the
   filters.

**Comment by dcdolson** [comment]
So where does the Filter-Rule AVP reside?

**Comment by yuvalif** [comment]
QoS-Final-Unit-Indication ::= < AVP Header: TBD18 >
                                   { Final-Unit-Action }
                                  ***[ Filter-Rule ]**
                                  *[ Filter-Id ]
                                   [ Redirect-Server-Extension ]
                                  *[ AVP ]

**Comment by dcdolson** [comment]
I was confused because Final-Unit-Action can appear in Final-Unit-Indication or QoS-Final-Unit-Indication.
I don't think the text is incorrect. Not sure if it could be made less confusing.

**Comment by yuvalif** [comment]
If you think that what i suggested does not clarify more, I'm fine with leaving the text as is.

**Comment by dcdolson** [comment]
Restriction-Filter-Rule is type IPFilterRule. It may have 'action' of 'permit' or 'deny'. Yet above wording says "In the case that Restriction-Filter-Rule AVP is used, all the packets matching the filters MUST be allowed."
Could it be that either permit or deny could be used, or is it intended that only 'permit' be used?

**Comment by dcdolson** [comment]
I tried rewriting it. Is the following accurate?
```
      The access device MUST restrict the user access according to the
      filter AVPs in the same group. See also section 5.6.3.
      In the case that Restriction-Filter-Rule AVP is used,
      all packets matching filters with 'permit' action MUST be allowed.
      In the case that Filter-Rule AVP is used, the Treatment-Action
      (E.g., drop, shape, mark, permit) MUST be applied to matching packets.
      In the case that Filter-Id AVP is used, all packets matching
      the specified filter MUST have the corresponding action applied,
      which may include other actions (e.g., QoS).
      In all cases, packets not matching any filters MUST be discarded.
```

**Comment by yuvalif** [comment]
section 8.36 do not give any info on whatever is inside the IPFilterRule AVP.
how about:

> 8.36.  Restriction-Filter-Rule AVP
> 
>    The Restriction-Filter-Rule AVP (AVP Code 438) is of type
>    IPFilterRule and provides filter rules corresponding to services that
>    are to remain accessible even if there are no more service units
>    granted.  The access device has to configure the specified filter
>    rules for the subscriber and MUST drop all the packets not matching
>    these filters.  Zero, one, or more such AVPs MAY be present in a
>    Credit-Control-Answer message or in an AA answer message.
>    **The action inside the IPFilterRule MUST be set to 'permit'.**

and then 8.35 does not need to get into these details:
The access device MUST restrict the user access according to the
      filter AVPs in the same group. See also section 5.6.3.
      In the case that Restriction-Filter-Rule AVP is used,
      all packets matching the filters MUST be allowed.
      In the case that Filter-Rule AVP is used, the Treatment-Action
      (E.g., drop, shape, mark, permit) MUST be applied to matching packets.
      In the case that Filter-Id AVP is used, all packets matching
      the specified filter MUST have the corresponding action applied,
      which may include other actions (e.g., QoS).
      In all cases, packets not matching any filters MUST be discarded.

**Comment by dcdolson** [comment]
I'm reluctant to put the description inside Restrict-Filter-Rule, in case it has been referenced by another standard. Would that be a concern?

**Comment by dcdolson** [comment]
Maybe going back to the RFC4006 wording, with minor change:

RESTRICT_ACCESS                 2
      The access device MUST restrict the user access according to the
      **filter AVPs in the same group: according to** IP packet filters defined in the Restriction-Filter-Rule AVP,
      **according to the packet classifier filters defined in Filter-Rule AVP,**
      or according to the ~~IP~~ packet filters identified by the Filter-Id
      AVP.  All the packets not matching **any** filters MUST be dropped
      (see section 5.6.3).

**Comment by lbertz02** [comment]
Per previous comments.

**Comment by dcdolson** [comment]
Fixed in commit 9c61dd1277c09ce14805fa225779c14d83728556
